src/Simplify_Call.cpp

```diff
@@ -125,28 +127,39 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
             return a;
         }
 
-        const bool shift_left = op->is_intrinsic(Call::shift_left);
         const Type t = op->type;
+        const bool b_pos = is_positive_const(b);
```

Comment:
From here to ~10 lines below is a little bit confusing because there's some redundant things happening. It took me quite a while to read it, I think it should be simplified somehow.

Some ideas:
- After checking is_const_int/is_const_uint, just determine b_pos and b_neg from the resulting constants, rather than calling is_positive_const/is_negative_const.
- Move this const_shift_left/const_shift_right stuff inside the first if below.

Reply:
You're right -- I cleaned this up a little. Please let me know what you think now!